src/Symfony/Component/TypeInfo/Type/BuiltinType.php

```diff
@@ -46,43 +40,13 @@ public function getTypeIdentifier(): TypeIdentifier
         return $this->typeIdentifier;
     }
 
-    public function isA(TypeIdentifier|string $subject): bool
+    public function isA(TypeIdentifier|string $expected): bool
     {
-        if ($subject instanceof TypeIdentifier) {
-            return $this->getTypeIdentifier() === $subject;
-        }
-
-        try {
-            return TypeIdentifier::from($subject) === $this->getTypeIdentifier();
-        } catch (\ValueError) {
-            return false;
-        }
-    }
-
-    /**
-     * @return self|UnionType<BuiltinType<TypeIdentifier::OBJECT>|BuiltinType<TypeIdentifier::RESOURCE>|BuiltinType<TypeIdentifier::ARRAY>|BuiltinType<TypeIdentifier::STRING>|BuiltinType<TypeIdentifier::FLOAT>|BuiltinType<TypeIdentifier::INT>|BuiltinType<TypeIdentifier::BOOL>>
-     */
-    public function asNonNullable(): self|UnionType
-    {
-        if (TypeIdentifier::NULL === $this->typeIdentifier) {
-            throw new LogicException('"null" cannot be turned as non nullable.');
-        }
-
-        // "mixed" is an alias of "object|resource|array|string|float|int|bool|null"
-        // therefore, its non-nullable version is "object|resource|array|string|float|int|bool"
```

Comment:
the removal of this logic is another proof that your assumption that making a non-nullable is simple without a dedicated API is wrong.